src/Hostnet/ruleset.xml

```diff
@@ -1,73 +1,332 @@
+    <arg name="extensions" value="php"/>
+    <arg name="colors"/>
+
+    <!-- Ignore warnings, show progress of the run and show sniff names -->
+    <arg value="ns"/>
+
+    <!-- Import PSR-2 coding standard (base) -->
+    <rule ref="PSR2"/>
+
+    <!-- Import all the Hostnet paths -->
+    <rule ref="HostnetPaths"/>
+
     <rule ref="Generic.CodeAnalysis.ForLoopShouldBeWhileLoop"/>
     <rule ref="Generic.CodeAnalysis.ForLoopWithTestFunctionCall"/>
     <rule ref="Generic.CodeAnalysis.JumbledIncrementer"/>
     <rule ref="Generic.CodeAnalysis.UnconditionalIfStatement"/>
-    <rule ref="Generic.CodeAnalysis.UnnecessaryFinalModifier"/>
-    <!--<rule ref="Generic.CodeAnalysis.UnusedFunctionParameter"/>-->
-    <rule ref="Generic.CodeAnalysis.UselessOverridingMethod"/>
 
     <rule ref="Generic.ControlStructures.InlineControlStructure"/>
 
-    <rule ref="Generic.Files.InlineHTML"/>
     <rule ref="Generic.Files.OneClassPerFile"/>
     <rule ref="Generic.Files.OneInterfacePerFile"/>
 
     <rule ref="Generic.Formatting.DisallowMultipleStatements"/>
-    <rule ref="Generic.Formatting.MultipleStatementAlignment"/>
 
     <rule ref="Generic.Functions.CallTimePassByReference"/>
     <rule ref="Generic.Metrics.CyclomaticComplexity"/>
     <rule ref="Generic.Metrics.NestingLevel"/>
 
-    <rule ref="Generic.NamingConventions.ConstructorName"/>
     <rule ref="Generic.NamingConventions.UpperCaseConstantName"/>
 
-    <rule ref="Generic.PHP.CharacterBeforePHPOpeningTag"/>
-    <rule ref="Generic.PHP.DeprecatedFunctions"/>
     <rule ref="Generic.PHP.ForbiddenFunctions"/>
     <rule ref="Generic.PHP.LowerCaseConstant"/>
     <rule ref="Generic.PHP.LowerCaseKeyword"/>
     <rule ref="Generic.PHP.NoSilencedErrors"/>
-    <rule ref="Generic.PHP.SAPIUsage"/>
 
-    <rule ref="Generic.Strings.UnnecessaryStringConcat">
+    <rule ref="Generic.WhiteSpace.DisallowTabIndent"/>
+    <rule ref="Generic.WhiteSpace.ScopeIndent"/>
+
+    <rule ref="Generic.Formatting.MultipleStatementAlignment"/>
+
+    <rule ref="Generic.Files.LineLength">
         <properties>
-            <property name="allowMultiline" value="true"/>
+            <property name="absoluteLineLimit" value="120"/>
         </properties>
     </rule>
 
-    <rule ref="Generic.WhiteSpace.DisallowTabIndent"/>
-    <rule ref="Generic.WhiteSpace.ScopeIndent"/>
+    <!-- Forbid `array(...)` -->
+    <rule ref="Generic.Arrays.DisallowLongArraySyntax"/>
+
+    <!-- Forbid duplicate classes -->
+    <rule ref="Generic.Classes.DuplicateClassName"/>
+
+    <!-- Forbid empty statements -->
+    <rule ref="Generic.CodeAnalysis.EmptyStatement"/>
 
-    <!-- Have 8 chars padding maximum and always show as errors -->
-    <rule ref="Generic.Formatting.MultipleStatementAlignment">
+    <!-- Forbid final methods in final classes -->
+    <rule ref="Generic.CodeAnalysis.UnnecessaryFinalModifier"/>
+
+    <!-- Forbid useless empty method overrides -->
+    <rule ref="Generic.CodeAnalysis.UselessOverridingMethod"/>
```

Comment:
Does this one take into account that the visibility _can_ change? I've seen this happen for a test once

Reply:
This should be allowed:

```php
class A
{
    private function foo()
    {

    }
}

class B extends A
{
    protected function foo()
    {

    }
}

class C extends B
{
    public function foo()
    {

    }
}
```

Reply:
The sniff does not complain about this.